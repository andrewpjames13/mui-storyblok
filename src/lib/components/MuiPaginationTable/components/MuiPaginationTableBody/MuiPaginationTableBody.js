import React, { createElement, useEffect } from 'react';
import { TableBody } from '@material-ui/core';
import PropTypes from 'prop-types';
import Storyblok from '../../../../utils/Storyblok';

import MuiPaginationTableRow from '../MuiPaginationTableRow/MuiPaginationTableRow';

const MuiPaginationTableBody = ({
  rootClass,
  content,
  handleRowNums,
  page,
  rowsPerPage,
}) => {
  const components = {
    MuiPaginationTableRow,
  };

  useEffect(() => {
    handleRowNums(content.length);
  }, []);

  const styles = Storyblok.arrayToMuiStyles(rootClass);

  console.log(content);

  return (
    <TableBody className={styles.root}>
      {content.slice(
        page * rowsPerPage,
        (page * rowsPerPage) + rowsPerPage,
      ).map((row, rowIndex) => createElement(
        components[row.component],
        Object.assign(row, { key: rowIndex }),
      ))
      }
    </TableBody>
  );
};

export default MuiPaginationTableBody;

MuiPaginationTableBody.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Mui Override or extend the styles applied to the component.
   */
  rootClass: PropTypes.arrayOf(PropTypes.string),
  /**
   * function
   * Sets the number of rows in table.
   */
  // eslint-disable-next-line react/require-default-props
  handleRowNums: PropTypes.func,
  /**
   * default: 0
   * page number of table.
   */
  // eslint-disable-next-line react/require-default-props
  page: PropTypes.number,
  /**
   * default: 10
   * Sets the number of rows in table.
   */
  // eslint-disable-next-line react/require-default-props
  rowsPerPage: PropTypes.number,
  /**
   * Content passed to render
   * components:
    MuiPaginationTableRow,
   */
  content: PropTypes.arrayOf(PropTypes.shape({
    component: PropTypes.string.isRequired,
  })).isRequired,
};

MuiPaginationTableBody.defaultProps = {
  rootClass: [],
};
