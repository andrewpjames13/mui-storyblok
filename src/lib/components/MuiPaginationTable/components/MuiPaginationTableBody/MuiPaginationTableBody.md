MuiPaginationTableBody Examples:

```js
const tBody = [
  {
    component: 'MuiPaginationTableRow',
    content: [
      {
        component: 'MuiTableCell',
        size: 'large',
        content: [
          {
            component: 'MuiTypography',
            content: [
              {
                component: 'MuiText',
                text: 'Table Body 1',
              },
            ],
          },
        ],
      },
      {
        component: 'MuiTableCell',
        size: 'large',
        content: [
          {
            component: 'MuiTypography',
            content: [
              {
                component: 'MuiText',
                text: 'Table Body 2',
              },
            ],
          },
        ],
      },
      {
        component: 'MuiTableCell',
        size: 'large',
        content: [
          {
            component: 'MuiTypography',
            content: [
              {
                component: 'MuiText',
                text: 'Table Body 3',
              },
            ],
          },
        ],
      },
    ],
  },
];

const handleRowNums = content => {
  return content.length;
}

<MuiPaginationTableBody
  content={tBody}
  handleRowNums={handleRowNums}
  rowsPerPage={10}
/>

```