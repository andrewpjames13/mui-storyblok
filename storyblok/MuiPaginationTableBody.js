module.exports = {
  "name": "MuiPaginationTableBody",
  "display_name": null,
  "updated_at": "2020-05-06T14:54:12.940Z",
  "schema": {
    "rootClass": {
      "type": "options",
      "description": "* stroyblok multiselect of css classes\n* Mui Override or extend the styles applied to the component."
    },
    "content": {
      "type": "bloks",
      "restrict_components": true,
      "component_whitelist": [
        "MuiPaginationTableRow"
      ]
    }
  },
  "image": null,
  "preview_field": null,
  "is_root": false,
  "preview_tmpl": null,
  "is_nestable": true,
  "all_presets": [],
  "preset_id": null,
  "real_name": "MuiPaginationTableBody",
  "component_group_uuid": null
}