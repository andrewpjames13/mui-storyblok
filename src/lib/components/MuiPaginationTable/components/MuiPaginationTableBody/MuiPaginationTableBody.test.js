import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import MuiPaginationTableBody from './MuiPaginationTableBody';

function setup() {
  const props = {
    content: [{
      hover: false,
      content: [{
        size: 'medium',
        align: 'center',
        content: [{
          align: 'inherit',
          color: 'inherit',
          width: '100%',
          height: '100%',
          content: [{
            text: 'table body 1',
            component: 'MuiText',
          }],
          variant: 'body1',
          component: 'MuiTypography',
        }],
        padding: 'default',
        component: 'MuiTableCell',
      }],
      selected: false,
      component: 'MuiTableRow',
    }],
  };
  const comp = shallow(<MuiPaginationTableBody {...props} />);
  return { comp, props };
}

describe('<MuiPaginationTableBody />', () => {
  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<MuiPaginationTableBody {...props} />);
    expect(tree).toMatchSnapshot();
  });

  it('should render MuiBody', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });
});
