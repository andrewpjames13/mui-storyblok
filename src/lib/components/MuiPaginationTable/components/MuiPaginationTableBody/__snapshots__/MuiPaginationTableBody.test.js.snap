// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MuiPaginationTableBody /> snapshot 1`] = `
<tbody
  className="MuiTableBody-root makeStyles-root-2"
/>
`;
